sentry/src/main/java/io/sentry/Span.java

```diff
@@ -317,4 +366,42 @@ public boolean isNoOp() {
   void setSpanFinishedCallback(final @Nullable SpanFinishedCallback callback) {
     this.spanFinishedCallback = callback;
   }
+
+  public void setStartDate(@NotNull SentryDate date) {
+    this.startTimestamp = date;
+  }
+
+  /**
+   * Updates the end date of the span. Note: This will only update the end date if the span is
+   * already finished.
+   *
+   * @param date the end date to set
+   * @return true if the end date was updated, false otherwise
+   */
+  public boolean updateEndDate(@NotNull SentryDate date) {
```

Comment:
`m`: I think we have to merge against `main`, because Stefano's changes landed, and this is actually part of the ISpan interface there